src/openfermion/transforms/_projection.py

```diff
@@ -0,0 +1,61 @@
+#   Licensed under the Apache License, Version 2.0 (the "License");
+#   you may not use this file except in compliance with the License.
+#   You may obtain a copy of the License at
+#
+#       http://www.apache.org/licenses/LICENSE-2.0
+#
+#   Unless required by applicable law or agreed to in writing, software
+#   distributed under the License is distributed on an "AS IS" BASIS,
+#   WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+#   See the License for the specific language governing permissions and
+#   limitations under the License.
+
+'''
+Functions to reduce the number of qubits involved
+in modelling a given system
+'''
+from openfermion.ops import QubitOperator
+import numpy
+
+
+def project_onto_sector(operator, qubits, sectors):
```

Comment:
isn't this almost identical to the routines for freezing orbitals in FermionOperator. Should we think about having this routine work for both FermionOperator and QubitOperator?

Reply:
Ah, thanks for pointing the freezing routine out, I hadn't been able to find it when I looked before.

The routines should be effectively identical, so combining them shouldn't be an issue (although I think we'll need different routines for both classes). But regarding terminology - to me it feels a bit odd talking about 'freezing orbitals' for a qubit system. What do you think? If you think it's fine I'll change it, otherwise perhaps we can find a name that seems obvious regardless of the type of operator.

As a general statement: given a Hermitian operator O with two eigenspaces, one can write any other operator A=B+C where [B,O]=0 and {C,O}=0, and then diagonalise B on either eigenspace of O, with error in the eigenvalues (or at least, the lowest eigenvalue) bounded by the Frobenius norm of C. If someone wanted to work out the code to do this it might be useful, but this sounds like a bit of a challenge.